### PR TITLE
[MISC] Fix secret keys

### PR DESCRIPTION
**Pull request by dragomirp**
Peer secrets were set as peer data.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -72,9 +72,9 @@ def __init__(self, *args):
             self,
             relation_name=PEER_RELATION_NAME,
             additional_secret_fields=[
-                AUTH_FILE_DATABAG_KEY,
-                CFG_FILE_DATABAG_KEY,
-                MONITORING_PASSWORD_KEY,
+                self._translate_field_to_secret_key(AUTH_FILE_DATABAG_KEY),
+                self._translate_field_to_secret_key(CFG_FILE_DATABAG_KEY),
+                self._translate_field_to_secret_key(MONITORING_PASSWORD_KEY),
```
Comment (dragomirp) [inline src/charm.py:77]: Needed the transformed peer data keys to properly register as secrets.

File: tests/integration/relations/pgbouncer_provider/test_pgbouncer_provider.py [hunk 2]
```diff
@@ -113,9 +113,16 @@ async def test_database_relation_with_charm_libraries(ops_test: OpsTest, pgb_cha
...
+            get_application_relation_data(
+                ops_test,
+                PGB,
+                PEER_RELATION_NAME,
+                "auth_file",
+            ),
         )
         assert secret_uri is not None
         assert password is None
+        assert auth_file is None
```
Comment (dragomirp) [inline tests/integration/relations/pgbouncer_provider/test_pgbouncer_provider.py:125]: Hopefully avoid the same footgun in the future.

File: src/relations/peers.py [hunk 3]
```diff
@@ -209,7 +209,6 @@ def _on_changed(self, event: HookEvent):
                 event.defer()
                 return
 
-            self.update_cfg(cfg)
```
Comment (dragomirp) [inline src/relations/peers.py:212]: Caused update secrets restart loop.

File: src/charm.py [hunk 4]
```diff
@@ -270,16 +270,7 @@ def _on_pgbouncer_pebble_ready(self, event: PebbleReadyEvent) -> None:
...
-        tls_enabled = all(self.tls.get_tls_files())
-        if self.model.relations.get("certificates", []) and not tls_enabled:
-            logger.debug(
-                "pgbouncer_pebble_ready: Deferring as certificates files are not yet populated for existing certificates relation"
-            )
-            self.unit.status = WaitingStatus("Waiting for certificates")
-            event.defer()
-            return
-        # in case of pod restart
-        elif tls_enabled:
```
Comment (dragomirp) [inline src/charm.py:282]: TLS test fails when deferring.

File: tests/unit/test_charm.py [hunk 5]
```diff
@@ -130,27 +130,6 @@ def test_on_pgbouncer_pebble_ready(self, _mkdir, _exec, _update_status, _):
...
-        self.assertEqual(initial_plan.to_yaml(), "{}\n")
-
-        container = self.harness.model.unit.get_container(PGB)
-        self.charm.on.pgbouncer_pebble_ready.emit(container)
-
-        assert not len(self.harness.model.unit.get_container(PGB).get_services())
-        get_tls_files.assert_called_once_with()
-        self.assertIsInstance(self.harness.model.unit.status, WaitingStatus)
-        self.assertEqual(self.harness.model.unit.status.message, "Waiting for certificates")
-
```
Comment (dragomirp) [inline tests/unit/test_charm.py:153]: No longer deferring in on_pebble_ready

**Review by juditnovak — APPROVED** [review]
LGTM

File: src/charm.py (hunk 1 above)
Comment (juditnovak) [reply src/charm.py:77]: Yes, but not just here...

`get/set_secret()` functions need to have it translated too, when passing the secret key to the lib.

**Review by juditnovak — CHANGES_REQUESTED** [review]

**Review by juditnovak — APPROVED** [review]
LGTM

**Review by marceloneppel — APPROVED** [review]
LGTM!

**Pull request merged**